### PR TITLE
Adds a global spark limit

### PR DESCRIPTION
**Pull request by lewcc**
## What Does This PR Do
Limits the total number of active spark effects to a high, yet hopefully reasonable, amount (500).

## Why It's Good For The Game
Sparks are *expensive*. We've had more than a few cases where spawning too many would have some pretty nasty consequences, so it'll be a good thing to put a bit of a cap on that.

## Testing
Broke all lights through the secrets tab, the spark limit was quickly exhausted (and it happened noticeably quicker).
Shot a few walls with a taser, noted that we still saw sparks.

## Changelog
:cl:
tweak: Added a global spark limit.
/:cl:

### CODE REVIEW

**Review by Contrabang — APPROVED** [review]
i like it (LGTM)

File: code/game/objects/effects/effect_system/effects_sparks.dm [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	// source - source of the sparks.
 
 	var/datum/effect_system/spark_spread/sparks = new
-	sparks.set_up(n, c, source)
+	sparks.set_up(sparks_allowed, c, source)
```
Comment (DGamerL) [inline code/game/objects/effects/effect_system/effects_sparks.dm:14]: Undefined var

**Review by DGamerL — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/effect_system/effects_sparks.dm (hunk 1 above)
Comment (lewcc) [reply code/game/objects/effects/effect_system/effects_sparks.dm]: addressed

**Review by DGamerL — APPROVED** [review]

**Comment by meow20** [comment]
sad day for plasma flooded turrets in ai sat.

**Review by S34NW — APPROVED** [review]

**Pull request merged**